tests/functional/test_hydro_infeasibilities_small_capacity.py

```diff
@@ -0,0 +1,385 @@
+    )
+
+    capacity = 2945
+    initial_level = 0.5 * capacity
+
+    expected_output_file = open(
+        "tests/functional/data/hydro_small_capacity/values-weekly.txt",
+        "r",
+    )
+    expected_output = expected_output_file.readlines()
+
+    week = 0
+    database.add_data(
+        "H",
+        "overall_target",
+        ConstantData(
+            float(expected_output[week + 7].strip().split("\t")[42])
+            - 0.75 * float(expected_output[week + 7].strip().split("\t")[43])
+        ),
+    )
+    database.add_data("H", "initial_level", ConstantData(initial_level))
+    problem = build_problem(
+        network,
+        database,
+        TimeBlock(1, list(range(168 * week, 168 * (week + 1)))),
+        1,
+    )
+    status = problem.solver.Solve()
+    assert status == problem.solver.INFEASIBLE
+
+
+def test_complete_year_as_weekly_blocks_with_hourly_infeasibilities() -> None:
+    """Solve weekly problems with heuristic weekly targets for the stock with modified inflow. Daily inflows remain the same. Inflows at the first hour of each day are large and there is oveflow that the heuristic didn't see due to agregation of data."""
+    inflow_data = (
+        np.loadtxt(
+            Path(__file__).parent
+            / "../../tests/functional/data/hydro_small_capacity/mod.txt",
+            usecols=0,
+        ).repeat(24)
+        / 24
+    )
+    variation_inflow = np.tile(np.array([2300] + [-100] * 23), 365)
+
+    database, network = create_database_and_network(
+        HYDRO_MODEL_WITH_TARGET,
+        return_to_initial_level=False,
+        bc=False,
+        inflow_data=list(inflow_data + variation_inflow),
+    )
+
+    capacity = 2945
+    initial_level = 0.5 * capacity
+
+    expected_output_file = open(
+        "tests/functional/data/hydro_small_capacity/values-weekly.txt",
+        "r",
+    )
+    expected_output = expected_output_file.readlines()
+
+    week = 0
+    database.add_data(
+        "H",
+        "overall_target",
+        ConstantData(
+            float(expected_output[week + 7].strip().split("\t")[42])
+            - 0.75 * float(expected_output[week + 7].strip().split("\t")[43])
+        ),
+    )
+    database.add_data("H", "initial_level", ConstantData(initial_level))
+    problem = build_problem(
+        network,
+        database,
+        TimeBlock(1, list(range(168 * week, 168 * (week + 1)))),
+        1,
+    )
+    status = problem.solver.Solve()
+    assert status == problem.solver.INFEASIBLE
+
+
+def create_database_and_network(
+    hydro_model: Model,
+    return_to_initial_level: bool,
+    bc: bool,
+    inflow_data: Optional[list[float]] = None,
+) -> tuple[DataBase, Network]:
+    capacity = 2945
+    initial_level = 0.5 * capacity
+    demand_data = np.loadtxt(
+        Path(__file__).parent
+        / "../../tests/functional/data/hydro_small_capacity/load.txt",
+        usecols=0,
+    )
+    if inflow_data is None:
+        inflow_data = list(
+            np.loadtxt(
```

Comment:
Is it possible to use your data reader defined in `src/hydro_heuristic/data.py` ?